frameworks/geniverse/resources/animation_core.js

```diff
@@ -2144,17 +2142,29 @@ sc_require('lib/burst-core');
     // Setup Draw-Loop
     ////////////////////////////////////////////////////////////////////////////
 
+    var $this = $(this);
+
     // cf. http://creativejs.com/resources/requestanimationframe/
     function drawLoop() {
-      timeoutID = setTimeout(function() {
-        requestID = requestAnimationFrame(drawLoop);
+      // use $.data() to store animation IDs with DOM node
+      $this.data('timeoutID', setTimeout(function() {
+        $this.data('requestID', requestAnimationFrame(drawLoop));
         // Drawing code goes here
         draw();
-      }, 1000 / fps);
+      }, 1000 / fps));
+    }
+
+    // cancel any previous animations
+    if ($this.data('requestID')) {
+      cancelAnimationFrame($this.data('requestID'));
+      $this.data('requestID', null);
     }
-    // don't start a timer if we've already got one
-    if (!timeoutID)
```

Comment:
This `$.geniverse()` function gets called every time the `retry` button is pressed. Every instantiation of the function is a new closure, so all closure variables are uninitialized. To save state from one instantiation to the next, we use `$.data()` to attach persistent state to the node.